src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,511 @@
+
+                if (value != _handshake)
+                {
+
+                }
+            }
+        }
+
+        internal int DataBits
+        {
+            set
+            {
+                Debug.Assert(!(value < MinDataBits || value > MaxDataBits), "An invalid value was passed to DataBits");
+                _dataBits = value;
+            }
+        }
+
+        internal Parity Parity
+        {
+            set
+            {
+                Debug.Assert(!(value < Parity.None || value > Parity.Space), "An invalid value was passed to Parity");
+
+                if (value != _parity)
+                {
+
+                }
+            }
+        }
+
+        internal StopBits StopBits
+        {
+            set
+            {
+                Debug.Assert(!(value < StopBits.One || value > StopBits.OnePointFive), "An invalid value was passed to StopBits");
+
+            }
+        }
+
+        internal bool DiscardNull
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+        internal byte ParityReplace
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+       // Uses Win32 method to dump out the receive buffer; analagous to MSComm's "InBufferCount = 0"
+        internal void DiscardInBuffer()
+        {
+        }
+
+        // Uses Win32 method to dump out the xmit buffer; analagous to MSComm's "OutBufferCount = 0"
+        internal void DiscardOutBuffer()
+        {
+        }
+
+       internal void SetBufferSizes(int readBufferSize, int writeBufferSize)
+        {
+            if (_handle == null) throw new IOException();
+            //IOCTL TIOCSSERIAL 14 max
+        }
+
+        internal bool IsOpen => _handle != null;
+
+
+        // Flush dumps the contents of the serial driver's internal read and write buffers.
+        // We actually expose the functionality for each, but fulfilling Stream's contract
+        // requires a Flush() method.  Fails if handle closed.
+        // Note: Serial driver's write buffer is *already* attempting to write it, so we can only wait until it finishes.
+        public override void Flush()
+        {
+            if (_handle == null) throw new ObjectDisposedException(SR.Port_not_open);
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new NotSupportedException(SR.NotSupported_UnseekableStream);
+        }
+
+        public override void SetLength(long value)
+        {
+            throw new NotSupportedException(SR.NotSupported_UnseekableStream);
+        }
+
+        public override int ReadByte()
+        {
+            return ReadByte(ReadTimeout);
+        }
+
+        internal unsafe int ReadByte(int timeout)
+        {
+            if (_handle == null)  throw new IOException();
+
+            selectList.Add(_handle);
+            Socket.Select(selectList, null, null, timeout);
```

Comment:
Is this needed to wait for data a the serial port? Anyway selectList.Add is called every time ReadByte but it seems that if it is removed from the list then a timeout exception is throw but then the list keeps growing on each success call? What happens with one that is added if Socket.Select throws?